app/main.py

```
@@ -1,3 +1,44 @@
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km: float = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: object) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: object) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: object) -> "Distance":
+        return Distance(self.km * other)
```

Issue: The __mul__ method multiplies self.km by 'other' without checking if 'other' is a numeric type. If 'other' is not a number, this will raise a runtime error. You should check the type and raise a TypeError for unsupported types, as required by the checklist.